tools/scripts/run_smoke_tests.py

```diff
@@ -37,14 +29,19 @@ def main():
     total_tests = 0
     all_results = defaultdict(list)
 
-    services=should_run_service(services)
+    service_ids = os.environ.get('AWS_SMOKE_TEST_SERVICE_IDS', '')
+    if not service_ids:
+        return 0
```

Comment:
i dont think this is what we want to do, this effectively creates a path where if `AWS_SMOKE_TEST_SERVICE_IDS` is removed for any reason we start ignoring smoke tests. this logic in the most abstract form should become

````
if (has service id AND service id is a known service in the SDK) {
  run _only_ the smoke test for service id
} else {
 run all tests
}
````

the makes it so we never skip smoke tests based on service id logic, we _must_ default to running all tests if we can make sense of _service_id_